# message with array of strings
string[] modelnames